iceoryx_hoofs/test/moduletests/test_posix_mutex.cpp

```diff
@@ -100,38 +122,63 @@ TEST_F(Mutex_test, CallingDestructorOnLockedMutexLeadsToTermination)
 }
 #endif
 
-TEST_F(Mutex_test, LockedMutexBlocksCallingThreadExecution)
+void tryLockReturnsFalseWhenMutexLockedInOtherThread(Mutex_test* test, iox::posix::mutex& mutex)
 {
-    std::atomic_bool isLockFinished{false};
-    sutNonRecursive.lock();
+    std::atomic_bool isTryLockSuccessful{true};
+    mutex.lock();
 
     std::thread lockThread([&] {
-        sutNonRecursive.lock();
-        isLockFinished.store(true);
-        sutNonRecursive.unlock();
+        isTryLockSuccessful = mutex.try_lock();
+        test->signalThreadReady();
     });
 
-    std::this_thread::sleep_for(std::chrono::milliseconds(500));
-    EXPECT_THAT(isLockFinished.load(), Eq(false));
+    test->waitForThread();
+    EXPECT_THAT(isTryLockSuccessful.load(), Eq(false));
 
-    sutNonRecursive.unlock();
+    mutex.unlock();
     lockThread.join();
+}
 
-    EXPECT_THAT(isLockFinished.load(), Eq(true));
+TEST_F(Mutex_test, TryLockReturnsFalseWhenMutexLockedInOtherThreadNonRecursiveMutex)
+{
+    tryLockReturnsFalseWhenMutexLockedInOtherThread(this, m_sutNonRecursive);
 }
 
-TEST_F(Mutex_test, TryLockWithRecursiveMutexReturnsFalseWhenMutexLockedInOtherThread)
+TEST_F(Mutex_test, TryLockReturnsFalseWhenMutexLockedInOtherThreadRecursiveMutex)
 {
-    std::atomic_bool isTryLockSuccessful{true};
-    sutRecursive.lock();
+    tryLockReturnsFalseWhenMutexLockedInOtherThread(this, m_sutRecursive);
+}
+
+void lockedMutexBlocks(Mutex_test* test, iox::posix::mutex& mutex)
+{
+    mutex.lock();
 
-    std::thread lockThread([&] { isTryLockSuccessful = sutRecursive.try_lock(); });
+    std::thread lockThread([&] {
+        test->signalThreadReady();
 
-    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+        auto start = std::chrono::high_resolution_clock::now();
+        mutex.lock();
+        mutex.unlock();
+        auto end = std::chrono::high_resolution_clock::now();
+        EXPECT_THAT(test->getDuration(start, end), Gt(test->WAIT_IN_NS));
+    });
 
-    EXPECT_THAT(isTryLockSuccessful.load(), Eq(false));
+    test->waitForThread();
 
-    sutRecursive.unlock();
+    std::this_thread::sleep_for(std::chrono::milliseconds(test->WAIT_IN_MS));
```

Comment:
Please use `std::chrono::milliseconds` for the `WAIT_IN_MS`/`WAIT_DURATION` then you can do a plain `EXPECT_THAT(end - start, Gg(test->WAIT_DURATION));`